bundle/bundle.go

```diff
@@ -44,11 +44,12 @@ const (
 
 // Bundle represents a loaded bundle. The bundle can contain data and policies.
 type Bundle struct {
-	Signatures SignaturesConfig
-	Manifest   Manifest
-	Data       map[string]interface{}
-	Modules    []ModuleFile
-	Wasm       []byte
+	Signatures  SignaturesConfig
+	Manifest    Manifest
+	Data        map[string]interface{}
+	Modules     []ModuleFile
+	Wasm        []byte // Deprecated. Use WasmModules instead
+	WasmModules []WasmModuleFile
```

Comment:
Since we've added support for multiple entrypoints inside of a single .wasm file I do wonder whether it's worth the complexity of supporting multiple modules inside the bundle. The cost of instantiating multiple VMs is high enough that we won't use this for the foreseeable future. Of course, moving from 1->N in the future would be harder so maybe we should just bite the bullet now. Just something to consider.

Reply:
Yea I have been eyeing that issue as I got this rebased on the changes to support >1 entrypoint per wasm module. Initially I was thinking to remove it... but I tend to agree that the cost of adding it later on would likely be higher than supporting it out of the gate. I still need to make the changes to have each wasm resolver support multiple entrypoints, will see if anything in that process changes my mind on it 😄